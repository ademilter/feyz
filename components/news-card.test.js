import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import NewsCard from './news-card'

const filter = (array) =>
  array.filter((row) => {
    return row.fields.createdDate && row.fields.public
  })

describe('NewsCard', () => {
  test('should render correctly.', () => {
    const mockData = filter(global.data.records)
    const article = mockData.filter((e) => !e.fields.tags.includes('Alıntı'))[0]
    const { asFragment, container } = render(
      <NewsCard id={article.id} {...article.fields} />
    )

    const {
      fields: { title, url, summary, image, author, tags }
    } = article
    const photo = image ? image[0].thumbnails?.large || image[0] : null
    const titleSelector = container.querySelector('h2 > a')
    const authorSelector = screen.getByText(new RegExp(author, 'i'))

    expect(asFragment()).toMatchSnapshot()
    expect(titleSelector).toHaveTextContent(title)
    expect(titleSelector).toHaveAttribute('href', url)
    expect(container.querySelector('p')).toHaveTextContent(summary)
    expect(container.querySelector('figure > a')).toHaveAttribute('href', url)
    expect(container.querySelector('picture > source')).toHaveAttribute(
      'srcset',
      photo.url
    )
    expect(authorSelector).not.toBeNull()
    expect(authorSelector).toHaveAttribute(
      'href',
      `https://twitter.com/${author}`
    )

    for (const tag of tags) {
      expect(screen.getByText(new RegExp(tag, 'i'))).not.toBeNull()
    }
  })

  test('should render correctly with "Alıntı" tag.', () => {
    const mockData = filter(global.data.records)
    const article = mockData.filter((e) => e.fields.tags.includes('Alıntı'))[0]
    const { asFragment, container } = render(
      <NewsCard id={article.id} {...article.fields} />
    )

    const {
      fields: { url, summary, author, tags }
    } = article
    const titleSelector = container.querySelector('h2 > a')
    const authorSelector = screen.getByText(new RegExp(author, 'i'))

    expect(asFragment()).toMatchSnapshot()
    expect(titleSelector).toHaveTextContent(summary)
    expect(titleSelector).toHaveAttribute('href', url)
    expect(authorSelector).not.toBeNull()
    expect(authorSelector).toHaveAttribute(
      'href',
      `https://twitter.com/${author}`
    )

    for (const tag of tags) {
      expect(screen.getByText(new RegExp(tag, 'i'))).not.toBeNull()
    }
  })
})
