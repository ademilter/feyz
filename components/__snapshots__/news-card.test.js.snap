// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard should have render with "Alıntı" tag. 1`] = `
<DocumentFragment>
  <article
    class=""
  >
    <h2>
      <a
        class=""
        href="http://alysha.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Unde sed at. Sed et velit consequatur sed.
      </a>
    </h2>
    <footer>
      <a
        aria-hidden="true"
        class=""
        href="http://alysha.com"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        11 months ago
      </a>
      , 
      <a
        class=""
        href="https://twitter.com/Blaise_Streich"
        rel="noopener noreferrer"
        target="_blank"
      >
        Blaise_Streich
      </a>
      , 
      <a
        href="/tag/yazilim"
      >
        Yazılım
      </a>
      , 
      <a
        href="/tag/turkce-kaynak"
      >
        Türkçe Kaynak
      </a>
      , 
      <a
        href="/tag/alinti"
      >
        Alıntı
      </a>
      , 
      <a
        href="/tag/yazi"
      >
        Blog Yazısı
      </a>
      , 
      <a
        href="/tag/podcast"
      >
        Podcast
      </a>
    </footer>
  </article>
</DocumentFragment>
`;

exports[`NewsCard should have render. 1`] = `
<DocumentFragment>
  <article
    class=""
  >
    <figure>
      <a
        aria-hidden="true"
        class=""
        href="https://chadd.net"
        rel="noopener noreferrer"
        tabindex="-1"
        target="_blank"
      >
        <picture>
          <source
            srcset="http://lorempixel.com/913/512"
          />
          <img
            alt="Doloremque animi officia."
            loading="lazy"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
          />
        </picture>
      </a>
    </figure>
    <div>
      <header>
        <h2>
          <a
            class=""
            href="https://chadd.net"
            rel="noopener noreferrer"
            target="_blank"
          >
            Doloremque animi officia.
          </a>
        </h2>
        <p>
          Eum facere ut expedita repudiandae iure labore laborum rerum. Dolorum autem ipsum similique iure error doloremque est. Hic voluptates assumenda nisi fugit dicta in.
        </p>
      </header>
      <footer>
        <a
          aria-hidden="true"
          class=""
          href="https://chadd.net"
          rel="noopener noreferrer"
          tabindex="-1"
          target="_blank"
        >
          3 months ago
        </a>
        , 
        <a
          class=""
          href="https://twitter.com/Elliott9"
          rel="noopener noreferrer"
          target="_blank"
        >
          Elliott9
        </a>
        , 
        <a
          href="/tag/frontend"
        >
          Frontend
        </a>
        , 
        <a
          href="/tag/turkce-kaynak"
        >
          Türkçe Kaynak
        </a>
        , 
        <a
          href="/tag/podcast"
        >
          Podcast
        </a>
        , 
        <a
          href="/tag/etkinlik"
        >
          Etkinlik
        </a>
      </footer>
    </div>
  </article>
</DocumentFragment>
`;
